### PR TITLE
Notice reports

### PR DESCRIPTION
**Pull request by Neterpila**
P.S. each commit has a description, so it's probably gonna be easier to go through them one by one

### CODE REVIEW

File: app/notice/notice.controller.js [hunk 1]
```diff
@@ -48,22 +48,40 @@ function add(req, res, next) {
...
-    if (!req.body) {
-        res.status(400).send({ message: "Content can not be empty!" });
-        return;
-    }
-    if(req.body.type){
-        return res
-                .status(400).send({message:'Can only edit description'})
-    }
+async function update(req, res) {
+    const notice = (await noticeService.get({ _id: req.params.id}, true, true))[0];
```
Comment (wojdzie) [inline app/notice/notice.controller.js]: Can you create a new service method to retrieve single element?

File: app/report/report.service.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+			total: (await noticeService.get({ type: "offer" }, true, true)).length,
+			active: (await noticeService.get({ type: "offer" })).length
+		}
+	}
+}
+
+async function periodic(start_date, end_date) {
+	return {
+		requests: {
+			total: (await Notice.find({
```
Comment (wojdzie) [inline app/report/report.service.js]: You should avoid calling DB 4 times with similar requests in one function. Try to get all required data with single DB call.

File: app/report/report.service.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
+const _ = require('lodash');
+const moment = require("moment");
+const noticeService = require('../notice/notice.service');
+const Notice = require('../notice/notice');
+
+async function overview() {
+	return {
+		total: (await noticeService.get({}, true, true)).length,
```
Comment (wojdzie) [inline app/report/report.service.js]: Same thing as below

**Review by wojdzie — CHANGES_REQUESTED** [review]
Please address identified issues

**Comment by Neterpila** [comment]
Identified issues addressed, please review

**Review by wojdzie — APPROVED** [review]

**Review by wojdzie — APPROVED** [review]

**Pull request merged**